platform/darwin/src/MGLMapSnapshotter.h

```diff
@@ -0,0 +1,105 @@
+#import <Foundation/Foundation.h>
+#import "MGLTypes.h"
+#import "MGLGeometry.h"
+#import "MGLMapCamera.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+MGL_EXPORT
+/**
+ The options to use when creating images with the `MGLMapsnapshotter`.
+ */
+@interface MGLMapSnapshotOptions : NSObject
+
+/**
+ Creates a set of options with the minimum required information
+ @param styleURL the style url to use
+ @param camera the camera settings
+ @param size the image size
+ */
+- (instancetype)initWithStyleURL:(NSURL*)styleURL camera:(MGLMapCamera*)camera size:(CGSize)size;
```

Comment:
Do we want to allow the style URL to be `nil` and default to Mapbox Streets, as MGLMapView and MGLOfflineStorage do?

Reply:
@1ec5 could you please help me understand why defaulting to Mapbox streets is a good option.

Reply:
When you create an MGLMapView or MGLOfflineRegion with a `nil` style URL, we default to the current version of Mapbox Streets:

https://github.com/mapbox/mapbox-gl-native/blob/53d072cbf539b30301fe3b77336751bf2c29af89/platform/ios/src/MGLMapView.mm#L397
https://github.com/mapbox/mapbox-gl-native/blob/53d072cbf539b30301fe3b77336751bf2c29af89/platform/darwin/src/MGLTilePyramidOfflineRegion.mm#L36

For consistency with these APIs, I think we should treat a `nil` style URL here the same way.